### ISSUE TITLE
Missing shape information for 'NonZero' node derived from 'Where' node

### ISSUE THREAD

**Issue by buddhapuneeth** [onnx/tensorflow-onnx#1203]
**Describe the bug**
Where operator is converted into NonZero and a Transpose operator ([ref](https://github.com/onnx/tensorflow-onnx/blob/3a633299202acb04e62d2bbe8ef71c2dc34d68cf/tf2onnx/onnx_opset/controlflow.py#L212)). But shape information is not added for NonZero node, which is causing issue while parsing the ONNX using ONNX C++ APIs.

**Urgency**
We are currently blocked on this for our internal project

**System information**
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Mac 
- Tensorflow Version:  2.2
- Python version: 3.6

**To Reproduce**
Describe steps/code to reproduce the behavior:
1. have a model with tf.where operator and convert into ONNX (using master branch of tf2onnx). During conversion, where will be decomposed into nonZero and transpose operators
2. Using ONNX C++ API, if you check for the shape of nonZero operator it will be empty.
```
onnx::ModelProto model;
ParseOnnx(input_stream, model)
const onnx::GraphProto& onnx_graph =  model.graph();
const OnnxValueInfos& value_infos = onnx_graph.value_info();
for(const auto& value_info : value_infos)
 {
     const onnx::TypeProto_Tensor& type_proto = value_info.type().tensor_type();
    if(!type_proto.has_shape())
                {
                  // error: shape is zero
                }
 }
```

**Expected behavior**
For example:
tf.where([[True, True], [False, True]])

outputs: [[0 0] [1 0] [ 1 1]] of shape 3*2

Internally in ONNX conversions we are doing a) NonZero which outputs [[0 1 1] [0 0 1]] of shape `2*3` and later transpose for the expected output of shape `3*2`.

In the conversion logic, shape of NonZero is missed.

**Screenshots**

**Additional context**
I am not allowed to share the original model. If needed, I will try to create a simple model with just where operator.

**Comment by TomWildenhain-Microsoft** [comment]
Is the NonZero node in the resulting ONNX model an input/output? I don't think we save the shape information into the final ONNX graph for nodes except for inputs/outputs.

**Comment by buddhapuneeth** [comment]
Yes. I realized the issue is with the shape inference of the intermediate tensors. I am tracking the issue using another ticket on ONNX: https://github.com/onnx/onnx/issues/3123.
So, closing this issue.
